### PR TITLE
Add configurable support for JCB (Apple Pay)

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary
Add configurable support for JCB Apple Pay Payment Network

## Motivation
https://jira.corp.stripe.com/browse/IOS-1212

## Testing
Added a unit test (STPApplePayTest.testJCBPaymentNetwork)

### CODE REVIEW

File: Stripe/PublicHeaders/STPAPIClient.h [hunk 1]
```diff
@@ -271,6 +274,13 @@ static NSString *const STPSDKVersion = @"15.0.0";
                                                    country:(NSString *)countryCode
                                                   currency:(NSString *)currencyCode;
 
+/**
+ Japanese users can enable JCB, Diners Club and Discover by calling this method with `YES`, after they have been approved by JCB.
+ 
+ The default value is NO.
+ */
++ (void)setJCBPaymentNetworkSupported:(BOOL)isSupported;
```
Comment (danj-stripe) [inline Stripe/PublicHeaders/STPAPIClient.h]: WDYT about using a class property instead? I think it gives better interop with Swift, but I'm not positive: https://useyourloaf.com/blog/objective-c-class-properties/

**Comment by yuki-stripe** [comment]
Made some changes, would appreciate another look @danj-stripe !

File: Tests/Tests/STPApplePayTest.m [hunk 2]
```diff
@@ -49,6 +49,9 @@ - (void)testCanSubmitPaymentRequestReturnsNOIfMerchantIdentifierIsNil {
 }
 
 - (void)testJCBPaymentNetwork {
+    if (&PKPaymentNetworkJCB == NULL) {
+        return;
```
Comment (danj-stripe) [inline Tests/Tests/STPApplePayTest.m]: If you wanted to be *really* complete you could instead test/assert that `setJCBPaymentNetworkSupported:YES` doesn't crash/add garbage/NULL to the `supportedPKPaymentNetworks` response.

Probably not worth it.

Comment (yuki-stripe) [reply Tests/Tests/STPApplePayTest.m]: Good idea!  `[Stripe supportedPKPaymentNetworks]` crashed in an earlier commit because I forgot this check.

**Pull request merged**